### PR TITLE
Add util pattern

### PR DESCRIPTION
**Pull request by AndresGarciadelaHuerta**
### Utils

To generate a util we use:

```
$ bundle exec rails g util Numbers clean double
```

This will generate the `NumbersUtil` class in the `app/utils` directory, as follows:

```ruby
class NumbersUtil < PowerTypes::BaseUtil

  def self.clean
    # Method code goes here
  end

  def self.double
    # Method code goes here
  end

end
```

And it will generate the spec file as well, in the `spec/utils` directory:

```ruby
require 'rails_helper'

describe NumbersUtil do
  describe '#clean' do
    pending 'describe what the util method clean does here'
  end

  describe '#double' do
    pending 'describe what the util method double does here'
  end

end
```

Every util will inherit from the class `PowerTypes::BaseUtil` which raises an error when the initialize method is called. The purpose of this is to ensure that all the utils methods work as class methods. Thus, there is no need to create an instance of the util to use its methods. For instance, we could use the `NumbersUtil` as follows:

```ruby
NumbersUtil.clean('5.000') # -> 5000
NumbersUtil.double(100) # -> 200
```

### CODE REVIEW

File: lib/power_types/patterns/base_util.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module PowerTypes
+  class BaseUtil
+    def initialize
+      raise PowerTypes::UtilError.new('a util cannot be instantiated')
```
Comment (ldlsegovia) [inline lib/power_types/patterns/base_util.rb:4]: *an* Util

**Review by ldlsegovia — APPROVED** [review]
👏🏻

File: lib/power_types/patterns/base_util.rb (hunk 1 above)
Comment (AndresGarciadelaHuerta) [reply lib/power_types/patterns/base_util.rb:4]: mm suena raro, diría que es una de las excepciones a la regla jaja [como estas](https://owl.purdue.edu/owl/general_writing/grammar/articles_a_versus_an.html#:~:text=an%20uprising-,Exceptions,-Use%20%22an)

Comment (ldlsegovia) [reply lib/power_types/patterns/base_util.rb:4]: Quería ver qué me contestabas porque no estaba seguro jaja. Cool 👍🏻

**Pull request merged**